### PR TITLE
Optimize dashboard UI for device with small width

### PR DESCRIPTION
**Pull request by guanbinrui**
persona_preview
setting_preview
dialog_preview

### CODE REVIEW

File: src/settings/settings.ts [hunk 1]
```diff
@@ -45,6 +45,10 @@ export const appearanceSettings = createGlobalSettings<Appearance>('apperance',
     primary: () => i18n.t('settings_appearance'),
 })
 
+export const responsiveUISettings = createGlobalSettings<boolean>('responsiveUI', webpackEnv.target === 'WKWebview', {
```
Comment (Jack-Works) [inline src/settings/settings.ts:48]: Why add a setting for it? It should be responsive by default 🤔

Comment (guanbinrui) [reply src/settings/settings.ts:48]: I changed my mind media queries are better than this manner.

File: src/extension/options-page/index.tsx [hunk 2]
```diff
@@ -135,7 +140,7 @@ function DashboardUI() {
                 <div className={classes.container}>
                     {loading ? null : (
                         <>
-                            <Drawer routers={routers} exitDashboard={null} />
+                            {xsMatched ? null : <Drawer routers={routers} exitDashboard={null} />}
```
Comment (Jack-Works) [inline src/extension/options-page/index.tsx:143]: Without a drawer (or something eq) how can people navigate in different tabs?

File: src/extension/options-page/index.tsx [hunk 3]
```diff
@@ -104,6 +98,7 @@ function DashboardUI() {
     const { t } = useI18N()
     const classes = useStyles()
     const history = useHistory<unknown>()
+    const xsMatched = webpackEnv.responsiveTarget === 'xs'
```
Comment (Jack-Works) [inline src/extension/options-page/index.tsx:101]: please use useMediaQuery

File: webpack.config.js [hunk 4]
```diff
@@ -263,6 +263,8 @@ module.exports = (argvEnv, argv) => {
         let firefoxVariant = undefined
         /** @type {'app' | 'browser' | 'facebookApp'} */
         let genericTarget = 'browser'
+        /** @type {'xs' | 'sm' | 'md' | 'lg' | 'xl'} */
+        let responsiveTarget = 'lg'
```
[the remark this answers is not in the document]
Comment (guanbinrui) [reply webpack.config.js:267]: The media query has a really uncertain behavior and brings unnecessary updates to UI. For example, it returns some default value at the first query and then returns the true value which we need of the rest queries.

File: src/extension/options-page/index.tsx (hunk 3 above)
Comment (guanbinrui) [reply src/extension/options-page/index.tsx:101]: Duplicate with https://github.com/DimensionDev/Maskbook/pull/1305#discussion_r456176158

File: webpack.config.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (guanbinrui) [reply webpack.config.js:267]: I found something in the source code ([useMediaQuery #L35](https://github.com/mui-org/material-ui/blob/45cb77ff02484e608467baded0b280fadd286467/packages/material-ui/src/useMediaQuery/useMediaQuery.js#L35)).

[the remark this answers is not in the document]
Comment (guanbinrui) [reply webpack.config.js:267]: The `responsiveTarget` flag can be coped as `perferResponsiveTarget` or `defaultResponsiveTarget` which will be the value of `defaultMatches` option when using the `useMediaQuery` hook.

File: src/env.d.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ declare const webpackEnv: {
     readonly target: 'Chromium' | 'Firefox' | 'WKWebview' | 'E2E' | undefined
     readonly firefoxVariant: 'android' | 'desktop' | 'GeckoView' | undefined
     readonly genericTarget: 'facebookApp' | 'browser'
-    readonly responsiveTarget: 'xs' | 'sm' | 'md' | 'lg' | 'xl'
+    readonly perferResponsiveTarget: 'xs' | 'sm' | 'md' | 'lg' | 'xl'
```
Comment (Jack-Works) [inline src/env.d.ts:9]: I suggest only 2 value, `preferResponsiveTarget: 'desktop' | 'mobile'`

Comment (guanbinrui) [reply src/env.d.ts:9]: Actually, we use a lots of CSS media queries like `[theme.breakpoints.down('xs')]]` in the codebase. I think it's easier to understand if we follow the same values in the JS even though we haven't used the values expected `xs`.

Comment (Jack-Works) [reply src/env.d.ts:9]: If you're not using these values, please comment them out at this moment.

**Pull request merged**